src/sync/generic_network_transport.hpp

```diff
@@ -123,11 +124,17 @@ std::error_code make_client_error_code(ClientErrorCode) noexcept;
 struct AppError {
 
     std::error_code error_code;
+    util::Optional<std::error_code> http_status_code;
```

Comment:
Does it make sense for this to be an error code as opposed to just int? The http error code is just an int value + the http category and the only place where we use the new field, we only read the int value.

Reply:
Yeah, its a bit overkill, I'll change it to an int.